source/Octopus.Client.Tests/Exceptions/OctopusExceptionFactoryFixture.cs

```diff
@@ -0,0 +1,304 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using Octopus.Client.Exceptions;
+
+namespace Octopus.Client.Tests.Exceptions
+{
+    public class OctopusExceptionFactoryFixture
+    {
+        [Test]
+        [TestCase(HttpStatusCode.BadRequest)]
+        [TestCase(HttpStatusCode.Conflict)]
+        public async Task Http400or409Response_WithNoPayLoad_ShouldCreateOctopusValidationException(
+            HttpStatusCode statusCode)
+        {
+            var httpResponseMessage = new HttpResponseMessage(statusCode);
+            var createdException = await OctopusExceptionFactory.CreateException(httpResponseMessage);
+
+            using (new AssertionScope())
+            {
+                createdException.Should().BeOfType<OctopusValidationException>();
+                createdException.HttpStatusCode.Should().Be((int)statusCode);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(CreatedOctopusExceptionTestData))]
+        public async Task CreatedOctopusValidationException_ShouldContainPayloadInformation(
+            OctopusExceptionFactory.OctopusErrorsContract octopusErrorsContract)
+        {
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
+
+            OctopusExceptionFactory.OctopusErrorsContract resultValidationObject = null;
+            if (HasAnyData(octopusErrorsContract))
+            {
+                var content = JsonConvert.SerializeObject(octopusErrorsContract);
+                httpResponseMessage.Content = new StringContent(content);
+                resultValidationObject = OctopusExceptionFactory.OctopusErrorsContractFromBody(content);
```

Comment:
This uses the logic under test to generate the expected result. This should come from an independent source.

Reply:
I don't believe this is actually testing anything, given the same code is being called to generate the _expected_ result and the _actual_ result. How could they not match?

An option would be to include in the test case a raw JSON payload and a structure containing the expected values we should see in the generated exception. This would make it clearer what is being tested, and give us more confidence that the result is what we expect.

Reply:
As discussed, the dependency on the factory is not the right approach as it will not provide safety net when there is a change in the factory. I've updated the implementation with `string payload, expected exception` pattern.